Fix misc. sheet issues, mess with individual character page structure, conditionals, etc.

PolomoPT opened the pull request:
I'll update this description soon. Will list everything I messed with that wasn't just visual (read: stuff that would be bad if messed up/that people need to know about)

Sorry Van. pls ask me about anything
😭 

Changes:
Any changes that are not simply aesthetic or have any relevance to people besides me and Van have the word "important" and/or are in bold


Aloy: The changes here are actually all aesthetic. I reformatted the Coil conditional but I think there are no implications whatsoever.

Amber: Added C4 "additional charges" header

Arataki Itto: Changes an st to stg ig

Baizhu: Added C1 Additional charges header

Beidou: +- important So C6 had to have a conditional for applying the Electro RES Shred. And the Burst had DMG Reduction in there, y'see, being applied regardless of whether it was on. So I made a conditional that appears regardless of constellation and applies the DMG Reduction, while additionally giving the Electro RES when at C6.

Bennett: Only aesthetic changes. Removed useless nodes/text being displayed when they shouldn't.

Candace: st -> chg

Chongyun: Made A1 ATK SPD have a Header

Collei: **IMPORTANT: added a conditional for C2 Energy Recharge.** Moved A1 to a Header within Skill

Cyno: **+- important: C1 used to display unconditionally, but that's no good. Since ATK SPD effect isn't used by optimizer, i just wiped it.** Other than that, moves headers a little and adds a few text nodes to the burst conditional, changes a few strings to chg.

Dehya: I wanted to use my fancy new stg and st so I changed text and added energy regen node to c4

Diluc: Changed his c2 cond from take dmg to get hit because that's how it works according to kqm tcl

Diona: Added nodes for c2 shields, header in c2, c2 effects as header in Skill.

Eula: Changed some node display order to match in-game. Changed ch to st.

Faruzan: **IMPORTANT: Added conditional for c6 because it doesn't snapshot while the regular buffs from Benefit do**

Fischl: **IMPORTANT: Changes C4 and C1 dmg nodes to be listed under the respective constellations (this WILL BREAK multi-opts!!). Adds headers for constellation effects.**

Ganyu: Added Additional charges header from 2

Hu Tao: Moved "CD" out of conditional. Change Activation Cost to st so it can be translated. Added Blood Blossom duration. Added incInterRes to C6.

Jean: syntax

Kazuha: syntax, moved A1 as header to Skill. Used new "effect duration" st's for his a4

Kaeya: Changed text display logic for c2 burst duration

Ayato: Moved Shunsuiken DMG out of conditional (conditional deleted). had to adjust a lot of canShows accordingly.

Keqing: moved c1 dmg as header to skill

Kirara: moved c2 shields into c2. (Did not fix https://github.com/frzyc/genshin-optimizer/issues/1193 yet, because I'm not sure if it's indeed an issue)

Klee: added "Charges | 2" to her Skill

Kujou Sara: her skill cond had a useless node (showing the ratio of increase). Changed a little logic of her skill conditional, but it shouldn't have any effects. Moved c6 and a4 effects to headers under Skill.

Kuki: Changed ch to st. Will be able to transfer translation, btw.

Layla: moved c1 shields to c1

Mika: fixed Soulwind duration percentage thing from bug reports

Mona: changed CritRateIncNeg to CritRateInc because math.abs already converts it to positive. Removed c4 conditional, moved the effects to burst. did a ch -> chg. Added duration to c1. Changed c6 unit to seconds.

Nahida: st -> stg. turned c4 into a team buff because it's EM. I dunno if I should've.

Nilou: Changed all references of "Watery Moon" to "Luminous Illusion" (will no longer break multi-opts). Added effect duration [element] to her c2 effects because i can.

Ningguang: added headers to passive and conditionals ability

Noelle: added headers. the way i did it for c6 attack increase meant i made a new node, alright?

Qiqi: **+- important: Turned A1 Healing Bonus into buff to active char**

Raiden: **Important: added conditional to her Burst that does... nothing, pre-C2, so I filled it with text nodes. post-c2, it fixes #1078** 

Razor: Changed text used by his wolf attacks so it makes sense. Added a few text nodes to his burst conditional.

Kokomi: moved her burst healing nodes out of conditional. Moved c4 and c6 effects and cond to Burst.

Sayu: **Changes:
Any changes that are not simply aesthetic or have any relevance to people besides me and Van have the word "important" and/or are in bold

Aloy: The changes here are actually all aesthetic. I reformatted the Coil conditional but I think there are no implications whatsoever.

Amber: Added C4 "additional charges" header

Arataki Itto: Changes an st to stg ig

Baizhu: Added C1 Additional charges header

Beidou: +- important So C6 had to have a conditional for applying the Electro RES Shred. And the Burst had DMG Reduction in there, y'see, being applied regardless of whether it was on. So I made a conditional that appears regardless of constellation and applies the DMG Reduction, while additionally giving the Electro RES when at C6.

Bennett: Only aesthetic changes. Removed useless nodes/text being displayed when they shouldn't.

Candace: st -> chg

Chongyun: Made A1 ATK SPD have a Header

Collei: IMPORTANT: added a conditional for C2 Energy Recharge. Moved A1 to a Header within Skill

Cyno: +- important: C1 used to display unconditionally, but that's no good. Since ATK SPD effect isn't used by optimizer, i just wiped it. Other than that, moves headers a little and adds a few text nodes to the burst conditional, changes a few strings to chg.

Dehya: I wanted to use my fancy new stg and st so I changed text and added energy regen node to c4

Diluc: Changed his c2 cond from take dmg to get hit because that's how it works according to kqm tcl

Diona: Added nodes for c2 shields, header in c2, c2 effects as header in Skill.

Eula: Changed some node display order to match in-game. Changed ch to st.

Faruzan: IMPORTANT: Added conditional for c6 because it doesn't snapshot while the regular buffs from Benefit do

Fischl: IMPORTANT: Changes C4 and C1 dmg nodes to be listed under the respective constellations (this WILL BREAK multi-opts!!). Adds headers for constellation effects.

Ganyu: Added Additional charges header from 2

Hu Tao: Moved "CD" out of conditional. Change Activation Cost to st so it can be translated. Added Blood Blossom duration. Added incInterRes to C6.

Jean: syntax

Kazuha: syntax, moved A1 as header to Skill. Used new "effect duration" st's for his a4

Kaeya: Changed text display logic for c2 burst duration

Ayato: Moved Shunsuiken DMG out of conditional (conditional deleted). had to adjust a lot of canShows accordingly.

Keqing: moved c1 dmg as header to skill

Kirara: moved c2 shields into c2. Fix #1193 

Klee: added "Charges | 2" to her Skill

Kujou Sara: her skill cond had a useless node (showing the ratio of increase). Changed a little logic of her skill conditional, but it shouldn't have any effects. Moved c6 and a4 effects to headers under Skill.

Kuki: Changed ch to st. Will be able to transfer translation, btw.

Layla: moved c1 shields to c1

Mika: **+ - important: added conditional to c6 effect as it may not be active even when Soulwind is.** fixed Soulwind duration percentage thing from bug reports.

Mona: changed CritRateIncNeg to CritRateInc because math.abs already converts it to positive. Removed c4 conditional, moved the effects to burst. did a ch -> chg. Added duration to c1. Changed c6 unit to seconds.

Nahida: st -> stg. turned c4 into a team buff because it's EM. I dunno if I should've.

Nilou: Changed all references of "Watery Moon" to "Luminous Illusion" (will no longer break multi-opts). Added effect duration [element] to her c2 effects because i can.

Ningguang: added headers to passive and conditionals ability

Noelle: added headers. the way i did it for c6 attack increase meant i made a new node, alright?

Qiqi: +- important: Turned A1 Healing Bonus into buff to active char

Raiden: Important: added conditional to her Burst that does... nothing, pre-C2, so I filled it with text nodes. post-c2, it fixes https://github.com/frzyc/genshin-optimizer/issues/1078

Razor: Changed text used by his wolf attacks so it makes sense. Added a few text nodes to his burst conditional.

Kokomi: moved her burst healing nodes out of conditional. Moved c4 and c6 effects and cond to Burst.

Sayu: **IMPORTANT: Her Press Kick DMG formula was benefitting from her C2 "time spent in windwheel" buff, while Kick DMG does not spend any time in windwheel. this should be wrong, so i removed that**. Otherwise, ch -> stg, removed a weird header.

Shenhe: Idk why her A4 nodes had greaterEqs set to A1, so I changed that. Made her C4 display in Stacks unit. Moved C1 extra charge to C1.

Heizou: ch, stg -> chg. Added a canShow parameter to his c6 so it doesn't display a 0% value.

Sucrose: effect duration [element], additional charges added to c1.

Thoma: st -> stg

Tighnari: Added a duration parameter to.. his a1, i think?

Anemo Traveler: I didn't know why it had separate burst displays for main and party; deleted. Added a few "duration"s

Dendro Traveler: stg -> chg ig

Electro Traveler: **+- important: i'm pretty sure their c4 effects weren't being accounted for so i added a few nodes for the different values.**

Geo Traveler: stg -> chg; ch -> st

Venti: his c2 additional shots used to be displaying prior to c2. they no longer do. added header too. st -> stg

Wanderer: moved his c6 to NAs

Xiangling: Moved A4 up to Skill. deleted c6 duration because it was wrong (used c1 duration)

Xiao: **IMPORTANT: Added conditional to c2 energy recharge.** Changed a bit of his Burst stuffs to show either in- or outside of the conditional. according to what was needed. Also did additional charges under c1.

Xingqiu: moved burst dmg nodes outside the conditional

Xinyan: Removed Shield Level conditional (moved nodes outside it)

Yae Miko: Added a few stuff from skillParams that wasn't there (i couldn't find out what some were. if someone knows, help). Added text header listings for the effects of c1, c2 and c6. c2 and c6 are under Skill. Moved c4 conditional to under Skill. Added Charges number.

Yanfei: I guess I mvoed a few headers/effects (passive, constellations) to their respective passives and constellations. I think I also added some headers when needed.

Yaoyao: new strings yay

Yelan: additional charges and added 's' unit to c6 duration.

Yoimiya: moved c6 to under NAs, adjusted some node order to match in-game.

Yun Jin: added headers to c2 and c6 effects

Zhongli: Moved A1 to under Skill. st -> stg.

Weapons:

Royal Series: changed strings

Calamity Queller: **IMPORTANT: Added "off-field" conditional and expected effects (more atk from passive)**. Changed conditional name to the effect name, Consummation.

Deathmatch: changed from string to math lango

Other index.tsx files:

Fixed whitespace in many, also updated Scanner files.

Added a few html tags for colored text. messed with Elemental Resonance conditionals to add said colored text.

Strings:

Added some stgs. Cleaned up some sts, while adding some sts too. Reworded a ton of stuff to be cleaner, and deleted anything unused. Some common strings were turned from chs to sts, don't fret as their translation can be salvaged.

Made some grammar tweaks to text throughout the website.** Otherwise, ch -> stg, removed a weird header.

Shenhe: Idk why her A4 nodes had greaterEqs set to A1, so I changed that. Made her C4 display in Stacks unit. Moved C1 extra charge to C1.

Heizou: ch, stg -> chg. Added a canShow parameter to his c6 so it doesn't display a 0% value.

Sucrose: effect duration [element], additional charges added to c1.

Thoma: st -> stg

Tighnari: Added a duration parameter to.. his a1, i think?

Anemo Traveler: I didn't know why it had separate burst displays for main and party; deleted. Added a few "duration"s

Dendro Traveler: stg -> chg ig

Electro Traveler: +- important: i'm pretty sure their c4 effects weren't being accounted for so i added a few nodes for the different values.

Geo Traveler: stg -> chg; ch -> st

Venti: his c2 additional shots used to be displaying prior to c2. they no longer do. added header too. st -> stg

Wanderer: moved his c6 to NAs

Xiangling: Moved A4 up to Skill. deleted c6 duration because it was wrong (used c1 duration)

Xiao: IMPORTANT: Added conditional to c2 energy recharge. Changed a bit of his Burst stuffs to show either in- or outside of the conditional. according to what was needed. Also did additional charges under c1.

Xingqiu: moved burst dmg nodes outside the conditional

Xinyan: Removed Shield Level conditional (moved nodes outside it)

Yae Miko: Added a few stuff from skillParams that wasn't there (i couldn't find out what some were. if someone knows, help). Added text header listings for the effects of c1, c2 and c6. c2 and c6 are under Skill. Moved c4 conditional to under Skill. Added Charges number.

Yanfei: I guess I mvoed a few headers/effects (passive, constellations) to their respective passives and constellations. I think I also added some headers when needed.

Yaoyao: new strings yay

Yelan: additional charges and added 's' unit to c6 duration.

Yoimiya: moved c6 to under NAs, adjusted some node order to match in-game.

Yun Jin: added headers to c2 and c6 effects

Zhongli: Moved A1 to under Skill. st -> stg.

Weapons:

Royal Series: changed strings

Calamity Queller: IMPORTANT: Added "off-field" conditional and expected effects (more atk from passive)

Deathmatch: changed from string to math lango

Other index.tsx files:

Fixed whitespace in many, also updated Scanner files.

Added a few html tags for colored text. messed with Elemental Resonance conditionals to add said colored text.

Strings:

Added some stgs. Cleaned up some sts, while adding some sts too. Reworded a ton of stuff to be cleaner, and deleted anything unused. Some common strings were turned from chs to sts, don't fret as their translation can be salvaged.

Made some grammar tweaks to text throughout the website.

nguyentvan7 commented:
Next time, you should make lots of small PRs rather than one mega PR. it makes it much easier to review and adjust

PolomoPT commented:
Changes:
_This will be a list of changes that were not simply aesthetic (strings or Header positions). Changes that have greater implications will also be highlighted_

Aloy: The changes here are actually all aesthetic. I reformatted the Coil conditional but I think there are no implications whatsoever.

Amber: **Added** C4 "additional charges" header

Arataki Itto: Changes an st to stg ig

Baizhu: **Added** C1 Additional charges header

Beidou: **Logic change:** So C6 had to have a conditional for applying the Electro RES Shred. And the Burst had DMG Reduction in there, y'see, being applied regardless of whether it was on. So I made a conditional that appears regardless of constellation and applies the DMG Reduction, while additionally giving the Electro RES when at C6.

Bennett: Only aesthetic changes. Removed useless nodes/text being displayed when they shouldn't.

Candace: st -> chg

Chongyun: Made A1 ATK SPD have a Header

Collei: `**Logic change:**` added a conditional for C2  Energy Recharge. Moved A1 to a Header within Skill

Cyno: **Kinda logic change:** C1 used to display unconditionally, but that's no good. Since ATK SPD effect isn't used by optimizer, i just wiped it. Other than that, moves headers a little and adds a **few text nodes to the burst conditional**, changes a few strings to chg.

Dehya: I wanted to use my fancy new stg and st so I changed text and **added energy regen node to c4**

Diluc: Changed his c2 cond from take dmg to get hit because that's how it works according to kqm tcl

Diona: **Added nodes for c2 shields**, header in c2, c2 effects as header in Skill.

Eula: Changed some node display order to match in-game. Changed ch to st.

Fischl: **IMPORTANT: Changes C4 and C1 dmg nodes to be listed under the respective constellations (this WILL BREAK multi-opts!!)**. Adds headers for constellation effects.

Ganyu: **Added** Additional charges header from 2

Hu Tao: Moved "CD" out of conditional. Change Activation Cost to st so it can be translated. Added Blood Blossom duration. Added incInterRes to C6.

Jean: syntax

Kazuha: syntax, moved A1 as header to Skill. Used new "effect duration" st's for his a4

Kaeya: Changed text display logic for c2 burst duration

Ayato: **Moved Shunsuiken DMG out of conditional (conditional deleted)**.  had to adjust a lot of canShows accordingly.

Keqing: moved  c1 dmg as header to skill

Kirara: moved c2 shields into c2. (Did not fix #1193 yet, because I'm not sure if it's indeed an issue)

Klee: added "Charges | 2" to her Skill

Kujou Sara: her skill cond had a useless node (showing the ratio of increase). Changed a little logic of her skill conditional, but it shouldn't have any effects. Moved c6 and a4 effects to headers under Skill.

Kuki: Changed ch to st. Will be able to transfer translation, btw.

Layla: moved c1 shields to c1

Mika: fixed Soulwind duration percentage thing from bug reports

Mona: changed CritRateIncNeg to CritRateInc because math.abs already converts it to positive. Removed c4 conditional, moved the effects to burst. did a ch -> chg. Added duration to c1. Changed c6 unit to seconds.

Nahida: st -> stg. **turned c4 into a team buff because it's EM. I dunno if I should've.**

Nilou: **IMPORTANT: Changed all references to "Watery Moon" to "Luminous Illusion". This breaks multi-opts and may affect databases, idk?** Added effect duration [element] to her c2 effects because i can.

Ningguang: added headers

Noelle: added headers. the way i did it for c6 attack increase meant i made a new node.

Qiqi: **+- important: Turned A1 Healing Bonus into team buff**

Raiden: **Important: added conditional to her Burst that does... nothing, pre-C2, so I filled it with text nodes. post-c2, it fixes #1078**

Razor: Changed text used by his wolf attacks so it makes sense. Added a few text nodes to his burst conditional.

PolomoPT commented:
In retrospect, I guess this does _not_ fix #1078. There's still the edge-case of weapon passive damage. but it's still pretty good

nguyentvan7 commented on apps/frontend/src/app/Data/Characters/Nahida/index.tsx:
```diff
@@ -523,6 +523,7 @@ const sheet: ICharacterSheet = {
         teamBuff: true,
         path: condC2QSAPath,
         value: condC2QSA,
+        canShow: equal(condC2Bloom, 'on', 1),
```
Technically these are not linked, since the quicken, etal buff has an 8s duration

nguyentvan7 commented on apps/frontend/src/app/Data/Characters/Nilou/index.tsx:
```diff
@@ -198,8 +198,8 @@ export const dmgFormulas = {
     whirl2Dmg: dmgNode('hp', dm.skill.whirl2Dmg, 'skill'),
-    moonDmg: dmgNode('hp', dm.skill.moonDmg, 'skill', {
-      premod: { skill_dmg_: c1_moon_dmg_ },
+    illusionDmg: dmgNode('hp', dm.skill.illusionDmg, 'skill', {
```
Breaks any multi opts. we can change the dm value, but maybe keep the first illusionDmg instance

nguyentvan7 commented on apps/frontend/src/app/Data/Characters/RaidenShogun/index.tsx:
```diff
@@ -105,12 +105,16 @@ const skillEye_ = equal(
   return dmgNode('atk', atkType, 'skill', {
     enemy: {
-      defIgn: greaterEq(input.constellation, 2, dm.constellation2.def_ignore),
+      defIgn: greaterEq(
```
This should be in the general premod so that all dmg instances are affected by this

PolomoPT replied:
I tried doing this. It's in my stashed changes; Idk if I did something wrong, but the damages didn't match when I tried running it. Gonna commit so you can tell me if I did anything wrong 😳

nguyentvan7 commented on apps/frontend/src/app/Data/Characters/Shenhe/index.tsx:
```diff
@@ -497,7 +489,16 @@ const sheet: ICharacterSheet = {
     passive2: ct.talentTem('passive2'),
     passive3: ct.talentTem('passive3'),
-    constellation1: ct.talentTem('constellation1'),
+    constellation1: ct.talentTem('constellation1', [
```
needs to be inside ct.fieldsTem

nguyentvan7 commented on apps/frontend/src/app/Data/Characters/Sucrose/index.tsx:
```diff
@@ -383,7 +383,16 @@ const sheet: ICharacterSheet = {
     ]),
     passive3: ct.talentTem('passive3'),
-    constellation1: ct.talentTem('constellation1'),
+    constellation1: ct.talentTem('constellation1', [
```
need ct.fieldstem

nguyentvan7 commented on apps/frontend/src/app/Data/Characters/Xiao/index.tsx:
```diff
@@ -366,12 +357,30 @@ const sheet: ICharacterSheet = {
     passive2: ct.talentTem('passive2'),
     passive3: ct.talentTem('passive3'),
-    constellation1: ct.talentTem('constellation1'),
+    constellation1: ct.talentTem('constellation1', [
```
ct.fieldstem

nguyentvan7 commented on apps/frontend/src/app/Data/Characters/YaeMiko/index.tsx:
```diff
@@ -267,33 +318,21 @@ const sheet: ICharacterSheet = {
     ]),
     passive3: ct.talentTem('passive3'),
-    constellation1: ct.talentTem('constellation1'),
+    constellation1: ct.talentTem('constellation1', [
```
ct.fieldstem

nguyentvan7 commented on apps/frontend/src/app/Data/Characters/Yelan/index.tsx:
```diff
@@ -432,7 +425,16 @@ const sheet: ICharacterSheet = {
     passive2: ct.talentTem('passive2'),
     passive3: ct.talentTem('passive3'),
-    constellation1: ct.talentTem('constellation1'),
+    constellation1: ct.talentTem('constellation1', [
```
ct.fieldstem and shouldnt this stay in the skill?

nguyentvan7 commented on apps/frontend/src/app/Data/Weapons/Polearm/CalamityQueller/index.tsx:
```diff
@@ -33,16 +33,9 @@ const dmg_Nodes = Object.fromEntries(
-    }),
-    constant(2, { name: trm('inactiveKey') })
-  ),
+  constant(1, {}),
```
the offfield flag is not used to change the calculation amount anywhere

PolomoPT commented on apps/frontend/src/app/Data/Characters/Albedo/index.tsx:
```diff
@@ -382,23 +383,6 @@ const sheet: ICharacterSheet = {
-      ct.condTem('burst', {
-        value: condBurstBlossom,
-        path: condBurstBlossomPath,
-        name: ct.ch('isotomaOnField'),
```
You mean conditional? Yeah. it's only used to display the dmg node. We want to pull those out of conditionals

PolomoPT commented on apps/frontend/src/app/Data/Characters/Amber/index.tsx:
```diff
@@ -399,7 +399,16 @@ const sheet: ICharacterSheet = {
-    constellation4: ct.talentTem('constellation4'),
+    constellation4: ct.talentTem('constellation4', [
+      {
+        fields: [
```
I wanted to try doing this as a standard with additional charges so as to unclutter the Headers under Skills and stuff. Idk, I think you could treat Additional Charges as either belonging to Skills or as common effects, like we do with Talent Boost.

Also, why do we need a ct.fieldsTem? Don't a lot of fieldsTems do this thing i did?

nguyentvan7 replied:
If you want to keep it under the constellation, then we should update the string to mention the skill at least "Additional Skill Charges" or something

PolomoPT replied:
![image](https://github.com/frzyc/genshin-optimizer/assets/121315943/84ce3ded-e6bb-41d7-8f15-88e5c50d3b68)
already did, ehe

PolomoPT commented on apps/frontend/src/app/Data/Characters/Cyno/index.tsx:
```diff
@@ -128,12 +128,6 @@ const a4_bolt_dmgInc = greaterEq(
   prod(percent(dm.passive2.bolt_dmgInc_), input.total.eleMas)
 )
 
-const c1_atkSPD_ = greaterEq(
```
Removed it all because ATK SPD isn't GO-relevant.

PolomoPT commented on apps/frontend/src/app/Data/Characters/Bennett/index.tsx:
```diff
@@ -383,12 +387,6 @@ const sheet: ICharacterSheet = {
         states: {
           activeInArea: {
             fields: [
-              {
```
I removed that because it's basically the multiplier — we don't display the multipliers for anything else. Same with C1 ratio.

PolomoPT commented on apps/frontend/src/app/Data/Characters/Faruzan/index.tsx:
```diff
@@ -359,19 +360,26 @@ const sheet: ICharacterSheet = {
-            text: ct.ch('c6Arrow'),
+        path: condC6CritPath,
+        value: condC6Crit,
+        name: ct.ch('baleCondName'),
```
I confused the two when setting up the `canShow` lmao.

PolomoPT commented on apps/frontend/src/app/Data/Characters/KujouSara/index.tsx:
```diff
@@ -251,17 +242,38 @@ const sheet: ICharacterSheet = {
+          },
+        ],
+      }),
+      ct.condTem('constellation6', {
```
no, because one snapshots while the other doesn't. I'd removed the conditional at one point, but then decided otherwise.

PolomoPT commented on apps/frontend/src/app/Data/Characters/Cyno/index.tsx:
```diff
@@ -384,68 +409,31 @@ const sheet: ICharacterSheet = {
-      }),
-    ]),
-
-    passive1: ct.talentTem('passive1', [
```
I think I'm gonna keep this the way I changed it to. I like having it under Skill because it's a new effect granted to the Skill.

PolomoPT commented on apps/frontend/src/app/Data/Characters/KujouSara/index.tsx:
```diff
@@ -284,34 +296,28 @@ const sheet: ICharacterSheet = {
             text: ct.chg('burst.skillParams.3'),
-            value: `${dm.burst.enerCost}`,
+            value: dm.burst.enerCost,
+            unit: 's',
```
lmao ig i figured since it did the ``${}`` it was also an `unit: s`

PolomoPT commented on apps/frontend/src/app/Data/Characters/KujouSara/index.tsx:
```diff
@@ -284,34 +296,28 @@ const sheet: ICharacterSheet = {
           {
             node: infoMut(dmgFormulas.constellation2.dmg, {
-              name: ct.chg(`skill.skillParams.0`),
+              name: st('dmg'),
```
That's an issue I've noticed with a few conditionals and these damage instances. It feels like it isn't perfect to use the generic strings (like `afterUse.burst` or `dmg`), but also like it's undeserving of a custom name. Maybe I can come back to this later? I feel like it's fine to keep it this way for now.

By the way, I think I mentioned this before, I changed the name because the old one was the Skill's damage name. And they aren't the same.

PolomoPT commented on apps/frontend/src/app/Data/Characters/Kirara/index.tsx:
```diff
@@ -470,7 +446,32 @@ const sheet: ICharacterSheet = {
     passive3: ct.talentTem('passive3'),
     constellation1: ct.talentTem('constellation1'),
-    constellation2: ct.talentTem('constellation2'),
+    constellation2: ct.talentTem('constellation2', [
```
oooh i now see the use of those.

PolomoPT commented on apps/frontend/src/app/Data/Characters/Beidou/index.tsx:
```diff
@@ -87,11 +88,11 @@ const dm = {
-// C6: During the duration of <b>Stormbreaker</b>
+// Burst: During the duration of <b>Stormbreaker</b>
 
-const [condC6Path, condC6] = cond(key, 'Constellation6')
```
any better way to do this? 🤔

nguyentvan7 commented on apps/frontend/src/app/Data/Weapons/Polearm/CalamityQueller/index.tsx:
```diff
@@ -33,7 +33,6 @@ const dmg_Nodes = Object.fromEntries(
   ])
 )
 const atkInc = prod(
```
Then change the premod to be premod: { atk: sum(atkInc, atkIncOffField) } }

nguyentvan7 replied:
Then change the offfield conditional to display atkIncOffField

PolomoPT commented on apps/frontend/src/app/Data/Characters/RaidenShogun/index.tsx:
```diff
@@ -546,6 +532,14 @@ const sheet: ICharacterSheet = {
         },
       }),
+      ct.headerTem('constellation2', {
+        canShow: equal(condInBurst, 'on', 1),
```
i think we do, because the def ignore is only active when in Burst, right?

nguyentvan7 commented on apps/frontend/src/app/Data/Characters/KujouSara/index.tsx:
```diff
@@ -250,7 +250,9 @@ const sheet: ICharacterSheet = {
               {
-                node: skillTenguAmbush_,
+                node: infoMut(skillTenguAmbush_disp, {
+                  name: st(`increase.atk`),
```
shouldnt this just use keymap.info('atk')?

PolomoPT replied:
ig so, then. did it to Bennett too.

nguyentvan7 commented on apps/frontend/src/app/Data/Weapons/Polearm/CalamityQueller/index.tsx:
```diff
@@ -32,18 +35,22 @@ const dmg_Nodes = Object.fromEntries(
-    condStacks,
-    objectKeyMap(range(1, 6), (i) => constant(i, { name: st('stacks') })),
-    0
+const atkInc = {
```
You don't need the {... } around it

nguyentvan7 commented on apps/frontend/src/app/Data/Characters/RaidenShogun/index.tsx:
```diff
@@ -277,6 +262,7 @@ export const data = dataObjForCharacterSheet(
       burstBoost: nodeC3,
       burst_dmg_: skillEye_,
       electro_dmg_: dmgFormulas.passive2.passive2ElecDmgBonus,
+      enemyDefIgn_: defIgn_
```
probably needs to be premod: { enemy: defIgn_ }

PolomoPT replied:
seems like it's indeed that, just defIgn rather than defIgn_

nguyentvan7 commented on apps/frontend/src/app/Data/Characters/Bennett/index.tsx:
```diff
@@ -90,34 +91,32 @@ const dm = {
   equal(input.activeCharKey, target.charKey, 1)
 )
+const activeInAreaAtkDisp = {
+  ...sum(burstAddlAtk, c1AddlAtk),
```
We separated these on purpose because the UI is confusing when combined. The C1 header indicates it is adding more atk, but its actually already factored into the calculation

PolomoPT replied:
So what I did was change the C1 listing to instead talk about the buff it gives. I think this should be clear enough.

It's gonna display like this: 
![image](https://github.com/frzyc/genshin-optimizer/assets/121315943/571a9226-6e31-43fa-9384-36f9f2da3b40)
tell me what you think :3

nguyentvan7 commented on apps/frontend/src/app/Data/Characters/KujouSara/index.tsx:
```diff
@@ -80,7 +81,8 @@ const atkIncRatio = subscript(
   condSkillTenguAmbush,
-  prod(input.base.atk, atkIncRatio)
+  prod(input.base.atk, atkIncRatio),
+  KeyMap.info('atk')
```
```suggestion
  { ...KeyMap.info('atk'), isTeamBuff: true }
```

nguyentvan7 commented on apps/frontend/src/app/Data/Weapons/Polearm/CalamityQueller/index.tsx:
```diff
@@ -60,8 +61,21 @@ const sheet: IWeaponSheet = {
     {
-      value: condStacks,
-      path: condStacksPath,
+      canShow: unequal(condStack, undefined, 1),
```
Move this conditional to be underneath the stack conditional

nguyentvan7 reviewed (changes requested):
Also fix your merge conflicts

nguyentvan7 commented on apps/frontend/src/app/Data/Characters/KujouSara/index.tsx:
```diff
@@ -275,6 +275,7 @@ const sheet: ICharacterSheet = {
         value: condC6,
         path: condC6Path,
         teamBuff: true,
+        canShow: equal(condSkillTenguAmbush, 'TenguJuuraiAmbush', 1),
```
If we are tying this buff to this conditional, then you also need to update the buff logic on line 94

nguyentvan7 commented on apps/frontend/src/app/Data/Characters/Mika/index.tsx:
```diff
@@ -121,7 +121,7 @@ const a1DetectorStacks_physical_dmg_ = equal(
   a1DetectorStacks_physical_dmg_disp
 )
-
+const [condC6CritPath, condC6Crit] = cond(key, 'c6Crit')
```
You're not using this for buff calculation

nguyentvan7 replied:
I don't think his C6 needs a cond actually, isn't the condition the same as inSoulWind?

PolomoPT replied:
yeah, but while the regular buff snapshots, CRIT from C6 does not[citation needed] (but it shouldn't, like other crit buffs)

PolomoPT replied:
oh, wait, i'm not sure if the regular buff snapshots

nguyentvan7 commented on libs/localization/assets/locales/en/sheet.json:
```diff
@@ -153,7 +150,9 @@
-    "skill_": "Ele. Skill Bonus Scaling"
+    "skill_": "Ele. Skill Bonus Scaling",
+    "atkInc": "ATK Inc. Bonus Scaling",
+    "percentStat": "{{percent}} {{stat}}"
```
Is this something that can work for all languages?

PolomoPT replied:
It could, in theory; turns out the issue is another, however: `[object Object]` 😭

nguyentvan7 replied:
Yes that is the issue with nesting translation, we had this issue before. Anyways, this percent stat thing shouldnt be like this. Instead we should be using a node with 
```
{
  text: st(stat.statName),
  value: percentValue
}
```

PolomoPT replied:
i'm fixing it. just took a break because i got involved with something

nguyentvan7 commented on apps/frontend/src/app/Data/Characters/Noelle/index.tsx:
```diff
@@ -402,7 +398,11 @@ const sheet: ICharacterSheet = {
+            text: st('bonusScaling.atkInc'),
+            value: st('bonusScaling.percentStat', {
+              percent: `${dm.constellation6.burstAtkBonus * 100}%`,
+              stat: 'DEF',
```
Untranslated string

nguyentvan7 commented on apps/frontend/src/app/Data/Characters/Bennett/index.tsx:
```diff
@@ -398,8 +398,11 @@ const sheet: ICharacterSheet = {
+            text: st('bonusScaling.atkInc'),
+            value: st('bonusScaling.percentStat', {
+              percent: `${dm.constellation1.atk_inc * 100}%`,
+              stat: 'Base ATK',
```
Untranslated string

PolomoPT commented on libs/dm-localization/src/lib/Data/sheet.ts:
```diff
@@ -84,12 +89,21 @@ const data = {
-    spread: 2191579811, // or maybe it is 982642368
+    spread: 2191579811,
+  },
+  stat: {
```
i only added the necessary things because i'm gonna work with this further in another PR. Will leave the bulk of changes until then.

nguyentvan7 commented on libs/localization/assets/locales/en/sheet.json:
```diff
@@ -152,7 +152,7 @@
   "bonusScaling": {
     "skill_": "Ele. Skill Bonus Scaling",
     "atkInc": "ATK Inc. Bonus Scaling",
-    "percentStat": "{{percent}} {{stat}}"
```
Do we really need a translated string for this? why not add the % ourselves in the sheet?

PolomoPT replied:
i think you commented on an old version of the file. Anyways, I'm actually doing this to give the option for translations to add connectives (pretty sure some languages will want to do the equivalent of adding an "of" in there.

uhhh i think i have other reasons but i can only think of this one right now and i think it's good enough
